Get NS_ERROR_WEBSOCKET_CONNECTION_REFUSED in Nomad 2.0.3

### Nomad version
Nomad v2.0.3
BuildDate 2026-06-09T17:33:24Z
Revision a2a5fee9c42d6481adcb9be865bcbccf8fd4d725

### Operating system and Environment details
cat /etc/os-release | head -2 ; uname -r
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
5.15.0-179-generic

### Issue
In UI when I want enter into container get this error immediately:
```
Select a task to start your session.
Customize your command, then hit ‘return’ to run.
$ nomad alloc exec -i -t -task utils-task 774dd5cf bash

The connection has closed.
```
In browser network console I see with error 500 and NS_ERROR_WEBSOCKET_CONNECTION_REFUSED (with IP the same). :
```
GET wss://nomad-master.service.consul:4646/v1/client/allocation/774dd5cf-6233-a74e-c34c-323ef24cc966/exec?task=utils-task&tty=true&ws_handshake=true&command=["bash"]
```
### Nomad Server logs:
journalctl -u nomad.service -f

`Jul 07 11:09:48 nomad-server1 nomad[1693057]:     2026-07-07T11:09:48.883Z [ERROR] http: request failed: method=GET path="/v1/client/allocation/774dd5cf-6233-a74e-c34c-323ef24cc966/exec?task=utils-task&tty=true&ws_handshake=true&region=global&command=%5B%22%2Fbin%2Fbash%22%5D" error="websocket connection not found in context" code=500`

### Reproduction steps
Click any job -> task name ->exec -> then enter on welcome screen

#### Expected Result
I can enter to the container

#### Actual Result
Get error `The connection has closed` in my 3 Nomad environments.

In 2.0.1 working fine.
Thanks!

--- a/.changelog/28234.txt
+++ b/.changelog/28234.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+ui: check websocket upgrade headers with multiple values
+```

--- a/command/agent/websockets.go
+++ b/command/agent/websockets.go
@@ -8,7 +8,6 @@ import (
 	"fmt"
 	"net/http"
 	"strconv"
-	"strings"
 
 	"github.com/gorilla/websocket"
 )
@@ -18,10 +17,9 @@ const (
 	ctxKeyWebSocketAuthToken = "ws_auth_token"
 )
 
-// isWebsocketUpgrade checks if the request is a websocket upgrade request
+// isWebsocketUpgrade checks if the request is a websocket upgrade request.
 func isWebsocketUpgrade(req *http.Request) bool {
-	return strings.EqualFold(req.Header.Get("Upgrade"), "websocket") &&
-		strings.EqualFold(req.Header.Get("Connection"), "upgrade")
+	return websocket.IsWebSocketUpgrade(req)
 }
 
 // wrapWebsocketHandler upgrades the HTTP connection to a websocket. Auditing

--- a/command/agent/websockets_test.go
+++ b/command/agent/websockets_test.go
@@ -220,3 +220,53 @@ func TestHTTP_ReadWsHandshake(t *testing.T) {
 		})
 	}
 }
+
+func TestIsWebsocketUpgrade_TokenListMismatch(t *testing.T) {
+	ci.Parallel(t)
+
+	cases := []struct {
+		name               string
+		connectionHeader   string
+		upgradeHeader      string
+		isWebsocketUpgrade bool
+	}{
+		{
+			name:               "standard upgrade",
+			connectionHeader:   "Upgrade",
+			upgradeHeader:      "websocket",
+			isWebsocketUpgrade: true,
+		},
+		{
+			name:               "comma-separated connection header",
+			connectionHeader:   "keep-alive, Upgrade",
+			upgradeHeader:      "websocket",
+			isWebsocketUpgrade: true,
+		},
+		{
+			name:               "not a websocket upgrade",
+			connectionHeader:   "keep-alive",
+			upgradeHeader:      "h2c",
+			isWebsocketUpgrade: false,
+		},
+		{
+			name:               "missing upgrade header",
+			connectionHeader:   "upgrade",
+			isWebsocketUpgrade: false,
+		},
+		{
+			name:               "missing connection header",
+			upgradeHeader:      "websocket",
+			isWebsocketUpgrade: false,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			req := httptest.NewRequest(http.MethodGet, "/v1/client/allocation/abc/exec", nil)
+			req.Header.Set("Connection", tc.connectionHeader)
+			req.Header.Set("Upgrade", tc.upgradeHeader)
+
+			must.Eq(t, tc.isWebsocketUpgrade, isWebsocketUpgrade(req))
+		})
+	}
+}

